### PR TITLE
[Fix #11629] Add new `Style/FileEmpty` cop

### LINKED ISSUE #11629: [New Cop Idea] File.empty?
Checking whether a file is empty in Ruby was always a bit weird:

- You can use `File.zero?` (class method of `File`)
- You can use `file.size == 0` or `file.size.zero?`

For some reason Ruby has always shied away from adding an `empty?` method to file objects (you can't do `file.empty?`, but at least Ruby 2.4 added an `File.empty?` method that is an alias of `File.zero?` and reads better. It'd be nice (and easy) to have some cop promoting the use of `File.empty?`.

### PR DESCRIPTION
**Pull request by ydah**
Fix #11629

```ruby
# bad
File.zero?('path/to/file')
File.size('path/to/file') == 0
File.size('path/to/file').zero?
File.read('path/to/file').empty?
File.binread('path/to/file').empty?
FileTest.zero?('path/to/file')

# good
File.empty?('path/to/file')
FileTest.empty?('path/to/file')
```

-----------------

Before submitting the PR make sure the following are checked:

* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Ran `bundle exec rake default`. It executes all tests and runs RuboCop on its own code.
* [x] Added an entry (file) to the [changelog folder](https://github.com/rubocop/rubocop/blob/master/changelog/) named `{change_type}_{change_description}.md` if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format) for details.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: lib/rubocop/cop/style/file_empty.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # Prefer to use File.empty?('path/to/file') when checking if a file is empty.
```
Comment (koic) [inline lib/rubocop/cop/style/file_empty.rb:6]:
```suggestion
      # Prefer to use `File.empty?('path/to/file')` when checking if a file is empty.
```

**Comment by ydah** [comment]
Thank you so much. I updated this PR.

File: lib/rubocop/cop/style/file_empty.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      # Prefer to use `File.empty?('path/to/file')` when checking if a file is empty.
+      #
+      # @safety
+      #   This cop's autocorrection is unsafe it because `File.size`, `File.read`,
+      #   and `File.binread` raise `ENOENT` exception when there is no file
+      #   corresponding to the path, while `File.empty?` does not raise an exception.
+      #
+      # @example
+      #   # bad
+      #   File.zero?('path/to/file')
```
Comment (bbatsov) [inline lib/rubocop/cop/style/file_empty.rb:15]: All those methods work with file objects as well, right?

Comment (ydah) [reply lib/rubocop/cop/style/file_empty.rb:15]: I think only `File.size` works with file objects.

```
irb(main):019:0> f
=> #<File:/foo/bar>
irb(main):020:0> f.zero?
(irb):20:in `<main>': undefined method `zero?' for #<File:/foo/bar> (NoMethodError)
        from /Users/ydah/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/irb-1.4.1/exe/irb:11:in `<top (required)>'
        from /Users/ydah/.rbenv/versions/3.1.2/bin/irb:25:in `load'
        from /Users/ydah/.rbenv/versions/3.1.2/bin/irb:25:in `<main>'
irb(main):021:0> f.zero?("foo")
(irb):21:in `<main>': undefined method `zero?' for #<File:/foo/bar> (NoMethodError)
        from /Users/ydah/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/irb-1.4.1/exe/irb:11:in `<top (required)>'
        from /Users/ydah/.rbenv/versions/3.1.2/bin/irb:25:in `load'
        from /Users/ydah/.rbenv/versions/3.1.2/bin/irb:25:in `<main>'
```

File: lib/rubocop/cop/style/file_empty.rb [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      #   File.size('path/to/file') == 0
```
[the remark this answers is not in the document]
Comment (ydah) [reply lib/rubocop/cop/style/file_empty.rb:16]: Indeed it is. Thank you very much. I added support for the negation substitution.

**Pull request merged**